client/src/App.vue

```diff
@@ -120,11 +121,21 @@ const fetchWorldList = () => {
       });
 };
 
+
 const plugWorldChat = async () => {
   worldChat = await wsClient.worldChatConnect(main.worldId);
   worldChat.onMessage((entry) => {
     const data = JSON.parse(entry);
     userFeed.publish(data.msg, data.name);
+    if (main.userId == data.id) {
+      const cmd = parseCommand(data.msg);
+
+      if (data.msg.startsWith('/tp ') && cmd) {
+        engine3d.user.position.x = cmd.x;
+        engine3d.user.position.y = cmd.y;
+        engine3d.user.position.z = cmd.z;
+      }
+    }
```

Comment:
Don't catch this here, as you are waiting on the feedback of the chat channel from the server for the command, in fact: the command shouldn't even be sent to the server anyway (at least not the one we are dealing with).

Take care of this in `handleSendChat` instead, if you detect `msg` is a command then don't even call `worldChat?.send(msg)` as we don't want to notify the server about this, but call `userFeed.publish(..., ...);` from there instead if you still want something to be displayed in the chat box (that what `userFeed` is for).
https://github.com/Blaxar/WideWorlds/blob/beb1cd3cc6e48db0c25d04428f94582d5bf27570/client/src/App.vue#L267C7-L267C21

Bonus thing is: you won't even need to check for the user ID anymore, because every thing you get there is solely the chat input from the local user.